scripts/listapranzo.js

```diff
@@ -85,6 +85,51 @@ module.exports = function (robot) {
     order.users[user.id] = dishes;
   };
 
+  var countMatches = function (str1, str2) {
+    var s1 = str1.split(' ');
+    var s2 = str2.split(' ');
+
+    s1 = s1.map(function (s) { return s.trim().toLowerCase(); });
+    s2 = s2.map(function (s) { return s.trim().toLowerCase(); });
+
+    var match = 0;
+
+    for (i = 0; i < s1.length; i++) {
+      for (j = 0; j < s2.length; j++) {
+        if (s1[i] === s2[j])
```

Comment:
In generale secondo me per essere accettata una ricerca deve contenere tutti i termini di ricerca (quale che sia la metodologia). Un match parziale va confermato perchè appunto non è detto sia quello che vuole l'utente (soprattutto visto che match ha delle conseguenze relativamente "reali" se uno non nota che la cosa selezionata è diversa da quella che aveva pensato di scrivere).

Per il sistema di match fuzzy, è un po' quello che hanno un sacco di sistemi di completamento directory/file. In ogni caso è solo il sistema di determinare se una entry è un match o no che cambia, dopo rimane uguale (i.e. se c'è più di un singolo match che soddisfa il sistema di matching usato i match vengono usati come alternative etc).

Il vantaggio del fuzzy è che puoi scrivere meno e soprattutto scrivere meno nella disambiguazione. Tipo se nel menù ci sono:

- spaghetti al pesto
- garganelli di portofino al pesto

col sistema a parole devi scrivere o "spaghetti pesto" o "garganelli/portofino pesto", e soprattutto se fai un typo fallisce/chiede conferma. Col fuzzy puoi scrivere "spapesto" e lo trova, e meno lettere si scrivono meno problemi di typo. Se non ti ispira fa niente cmq, è comunque questione di preferenze.